mpxe/sims/sim.py

```diff
@@ -0,0 +1,19 @@
+from mpxe.protogen import stores_pb2
+from mpxe.protogen import gpio_pb2
+
+class Sim:
+    def handle_update(self, pm, proj):
+        pass
+
+    def handle_log(self, pm, proj, log):
+        print('[{}]'.format(proj.name), log)
+
+    def set_gpio(self, proj, port, pin, state):
+        ind = (ord(port.capitalize()) - ord('A')) * 16 + pin
+        gpio_msg = gpio_pb2.MxGpioStore()
+        gpio_msg.state.extend([0] * 6 * 16)
+        gpio_msg.state[ind] = state
+        gpio_mask = gpio_pb2.MxGpioStore()
+        gpio_mask.state.extend([0] * 6 * 16)
+        gpio_mask.state[ind] = state
+        proj.write_store(stores_pb2.MxStoreType.GPIO, gpio_msg, gpio_mask)
```

Comment:
I think there needs to be some utility abstractions to make accessing each type of store easier, like this but in another module other than `sim` since this class will get too large. We should have utility modules for accessing gpio, adc, etc - one for each major type of store, so that people writing sims / integration tests don't have to deal with the low level store stuff.

For example, there could be an Ads1015 utility class where one instance corresponds to one instance of the ADS1015 driver once we figure out multiple stores; I think the key stuff in project.py is meant for this. Then the PedalBoard sim could just provide access to an Ads1015 utility class. (Maybe even structure each utility class as a Sim so you could have some sort of hierarchy of sims - a PedalBoard sim has an Ads1015 sim, etc.) This might be best for another ticket though.